DependencyInjection/Configuration.php

```diff
@@ -34,9 +34,13 @@ public function getConfigTreeBuilder()
                     ->useAttributeAsKey('name')
                     ->prototype('array')
                         ->children()
-                            ->scalarNode('credentials')
-                                ->info('Path to the project\'s Service Account credentials file. If omitted, the credentials will be auto-dicovered as described in https://firebase-php.readthedocs.io/en/stable/setup.html#with-autodiscovery')
-                                ->example('%kernel.project_dir%/config/my_project_credentials.json')
+                            ->variableNode('credentials')
+                                ->info('Path to the project\'s Service Account credentials file or the json/array credentials parameters. If omitted, the credentials will be auto-dicovered as described in https://firebase-php.readthedocs.io/en/stable/setup.html#with-autodiscovery')
+                                ->example('%kernel.project_dir%/config/my_project_credentials.json or credentials: type ..')
+                                ->validate()
+                                    ->ifTrue(function ($v) {return false === is_string($v) && false === is_array($v);})
```

Comment:
```suggestion
                                    ->ifTrue(function ($v) {return !is_string($v) && !is_array($v);})
```

I often like being more explicit more as well, but in this case I think the "short" checks would be fine as well.